blaze/mahimahi/server/filestore.py

```diff
@@ -10,22 +10,50 @@
 from blaze.util import encoding
 
 
+DATE_HEADER = "date"
+PRAGMA_HEADER = "pragma"
+EXPIRES_HEADER = "expires"
+LAST_MODIFIED_HEADER = "last-modified"
 CACHE_CONTROL_HEADER = "cache-control"
 TRANSFER_ENCODING_HEADER = "transfer-encoding"
 ACCESS_CONTROL_ALLOW_ORIGIN_HEADER = "access-control-allow-origin"
 REMOVE_HEADERS = [
-    "cache-control",
     "connection",
     "content-length",
-    "expires",
     "keep-alive",
-    "last-modified",
-    "date",
     "age",
     "etag",
+    CACHE_CONTROL_HEADER,
+    DATE_HEADER,
+    EXPIRES_HEADER,
+    LAST_MODIFIED_HEADER,
+    PRAGMA_HEADER,
+    TRANSFER_ENCODING_HEADER,
 ]
 
 
+def check_cacheability(headers: Dict[str, str]) -> bool:
+    cache_control = headers.get(CACHE_CONTROL_HEADER, "")
+    if "no-cache" in cache_control or "no-store" in cache_control:
+        return False
+
+    if "max-age=" in cache_control:
+        try:
+            max_age = int(cache_control.split("max-age=")[1].split(",")[0])
```

Comment:
what if cache-control header is not set but expires  or last-modified is non-zero/in the future. 
https://github.com/muralisr/mahimahi/blob/master/src/frontend/is_cacheable.cc (lines 124 to 164)

Reply:
If there is a `max-age=0`, we say it's not cacheable according to line 118 of the `is_cacheable.cc`. If there's no value or its set to an invalid value, then it passes and checks expires and last-modified in the last line if the function. If the `max-age` is greater than zero, we say its cacheable (line 159 of `is_cacheable.cc`)